Processing recommendations
==========================

Currently, Qiita supports the processing of raw data from:

#. Target gene barcoded sequencing
#. Shotgun sequencing

Note that the selected processing are mainly guided so we can perform meta-analyses, this is combine different studies,
even from different wet lab techniques or sequencing technologies.


Target gene barcoded sequencing
-------------------------------

For this you can start with raw, not demultiplexed data or per_sample_FASTQ. Either way, you will need to
"Split libraries and QC", which uses the default in QIIME 1.9.1. Once your demultiplexed and QCed artifact is created
you need to select which processing to perform. There are two main ideologies/methodologies to process target
gene data: sequence clustering and sequence cleanup.

Sequencing deblur (preferred)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For this we use `deblur <https://github.com/biocore/deblur>`_. Here 2 BIOM tables are generated by default: fina.biom and final.only-16s.biom. The former is the full biom table, which can be used with any target gene and wetlab work;
the latter is the trimmed version to those sequences that match Greengenes at 80% similarity, a really basic and naive filtering. Each of those BIOM tables, is accompanied by a FASTA that contains
the representative sequences. The OTU IDs are given by the unique sequence.

Note that deblur needs all sequences to be trimmed at the same length, thus the recommended pipeline is to trim everything at 150bp and the deblur.

Sequencing clustering
^^^^^^^^^^^^^^^^^^^^^

Here we use close reference picking, for an explanation of the different picking methods see
`"Subsampled open-reference clustering creates consistent, comprehensive OTU definitions and scales to billions of sequences" <https://peerj.com/articles/545/>`_.
Here we generate a single BIOM table with the OTUs/per-sample. The OTU IDs are given based on the reference database selected.

Currently, we have the reference databases: Greengenes version 3_8-97, Silva 119 and Unite 7. Depending on your selection is if the reference has a phylogenetic tree.

Shotgun sequencing
------------------

Qiita currently has one shotgun metagenomics data analysis pipeline: `Shogun <https://msystems.asm.org/content/3/6/e00069-18>`_.

The current workflow is as follows:

#. Removal of adapter sequence and quality control: `Atropos <https://github.com/jdidion/atropos/>`_
#. Host contamination using `Bowtie2 <http://bowtie-bio.sourceforge.net/bowtie2/index.shtml>`_
#. Taxonomy profiling using three different aligners and two different reference databases; see sections below

Note that we recommend only uploading sequences that have already been through QC and human sequence removal. However, we
recommend that all sequence files go through adapter and quality control within the system to ensure they are ready for
subsequent analyses. Currently, the command removes adaptor sequences (only KAPA HyperPlus with iTru).

For host removal we currently support Danio Rerio, Drosophila Melanogaster, Mus Musculus, PhiX, and Rattus Norvegicus.

Note that the Shogun command produces 4 output artifacts:
- The Alignment Profile BIOM artifact, which contains the alignment files
- A Taxonomic Prediction - phylum BIOM artifact, which contains the taxonomic predictions based on the alignment
- A Taxonomic Prediction - genera BIOM artifact, which contains the taxonomic predictions based on the alignment
- A Taxonomic Prediction - species BIOM artifact, which contains the taxonomic predictions based on the alignment
The 3 Taxonomic Prediction files can be used for subsequent analysis and visualization.

Shogun aligners
^^^^^^^^^^^^^^^

[ToDo]

Shogun reference databases
^^^^^^^^^^^^^^^^^^^^^^^^^^

#. WoLr1 ("Web of Life" release 1)
   An even representation of microbial diversity, selected using an prototype selection algorithm based
   on the MinHash distance matrix among all non-redundant bacterial and archaeal genomes from NCBI
   (RefSeq and GenBank, complete and draft), plus several genome quality control criteria. A
   high-quality reference phylogeny is available for this genome pool, enabling subsequent
   phylogeny-based analyses. Also available are curated taxonomic annotations, based on NCBI and GTDB
   systems.
   - Domains: Bacteria, Archaea
   - Number of genomes: 10,575
   - Total length (bp): 32,861,886,373
   - Numbers of taxonomic units:

     - Kingdoms: 2
     - Phyla: 146
     - Classes: 89
     - Orders: 196
     - Families: 422
     - Genera: 2,081
     - Species: 9,105
     - Strains: 89
     - Note: Nucleotide sequences per genome were concatenated with a linker of 20 "N"s.

   - Citation: Zhu Q, Mai U, Pfeiffer W, et al. Phylogenomics of 10,575 genomes reveals evolutionary
     proximity between domains Bacteria and Archaea. Nat Commun. 2019. 10(1):5477. doi: 10.1038/s41467-019-13443-4.

#. Rep94
   NCBI representative and reference microbial genomes, corresponding to RefSeq release 94.
   - Domains: Bacteria, Archaea
   - Number of genomes: 5,808
   - Total length (bp): 23,165,526,011
   - Numbers of taxonomic units:

     - Kingdoms: 2
     - Phyla: 38
     - Classes: 85
     - Orders: 186
     - Families: 427
     - Genera: 1,931
     - Species: 5,636
     - Strains: 84

   - Note: Nucleotide sequences per genome were concatenated with a linker of 20 "N"s.

#. Rep82
   NCBI representative and reference microbial genomes, corresponding to RefSeq release 82.
   - Domains: Bacteria, Archaea, Viruses/Viroids
   - Number of genomes: 10,519
   - Total length (bp): 20,387,349,319
   - Numbers of taxonomic units:

     - Kingdoms: 6
     - Phyla: 55
     - Classes: 362
     - Orders: 182
     - Families: 452
     - Genera: 2,264
     - Species: 11,852
     - Strains: 4,263

   - Note: Plasmids were isolated from bacterial and archaeal host genomes and considered as separate genomes.
